lms/static/js/ccx/async_grade_book_download.js

```diff
@@ -0,0 +1,137 @@
+var defineJs = window.define || RequireJS.define;
+
+defineJs("async_grade_book_download",
+  [
+    'backbone',
+    'underscore',
+    'gettext'
+  ],
+  function (Backbone, _, gettext) {
+    'use strict';
+    var GradeBookDownloadView = Backbone.View.extend({
+
+      events: {
+        "click a#ccx-calculate-grades-csv" : "asyncPrepareGradeBook"
+      },
+
+      initialize: function() {
+        // set interval to fetch list of files that are ready for download.
+        this.clearMessageBoxes();
+        this.showReadyGradeBooks();
+      },
+
+      render: function () {
+        return this;
+      },
+
+      reloadDownloadList: function() {
+        // refresh list of grade books, ready for download.
+        var self = this;
+        this.clearMessageBoxes();
+        // Fetch interval for grade book download request 40 sec.
+        var POLL_INTERVAL = 20000;
```

Comment:
it is logic i copied from https://github.com/edx/edx-platform/blob/master/lms/static/coffee/src/instructor_dashboard/util.coffee#L335.
Because when we prepare gradebook it takes some time and then ready gradebooks are save on S3. and after 20s we check if gradebook is ready then display it.